src/Microsoft.Extensions.Configuration.Abstractions/ConfigurationExtensions.cs

```diff
@@ -26,17 +26,28 @@ public static string GetConnectionString(this IConfiguration configuration, stri
         /// </summary>
         /// <param name="configuration">The <see cref="IConfiguration"/> to enumerate.</param>
         /// <returns>An enumeration of key value pairs.</returns>
-        public static IEnumerable<KeyValuePair<string, string>> AsEnumerable(this IConfiguration configuration)
+        public static IEnumerable<KeyValuePair<string, string>> AsEnumerable(this IConfiguration configuration) => configuration.AsEnumerable(removePathFromChildKeys: false);
+
```

Comment:
@HaoK The best names I have come up with are `includePathFromRoot` (true by default), `trimPathFromRoot` (false by default) and `useRelativePaths` (false by default). Naming is hard. I will give it some sleep :smile: